File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> freq;
        int maxi = INT_MIN;
        int maxi_num = nums[0];
        for(int num: nums)
        {
            if(maxi < ++freq[num])
            {
                maxi = freq[num];
                maxi_num = num;
            }
        }
        return maxi_num;
    }
};

File: take-u-forward/169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ele = nums[0];
        int count = 1;
        for(int ind=1; ind < nums.length; ind++) {
            if(count == 0)
                ele = nums[ind];
            count += nums[ind]==ele ? 1 : -1;
        }
        return ele;
    }
}